aidial_adapter_openai/images.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, AsyncGenerator, Optional
+
+import aiohttp
+from fastapi.responses import JSONResponse, Response, StreamingResponse
+
+from aidial_adapter_openai.utils.exceptions import HTTPException
+from aidial_adapter_openai.utils.storage import FileStorage, upload_base64_file
+from aidial_adapter_openai.utils.streaming import (
+    END_CHUNK,
+    build_chunk,
+    chunk_format,
+    generate_id,
+)
+
+
+async def generate_image(api_url: str, api_key: str, user_prompt: str) -> Any:
+    async with aiohttp.ClientSession() as session:
+        async with session.post(
+            api_url,
+            json={"prompt": user_prompt, "response_format": "b64_json"},
+            headers={"api-key": api_key},
+        ) as response:
+            status_code = response.status
+
+            data = await response.json()
+
+            if status_code == 200:
+                return data
+            else:
+                return JSONResponse(content=data, status_code=status_code)
+
```

Comment:
It's would be more straighforward to raise an exception here instead of returning JSONResponse. There is a FastAPI exception handler, which will turn it eventually into a proper JSONResponse.

Reply:
You could raise `aidial_adapter_openai.utils.exceptions.HTTPException`.

Alternatively, did you try to use OpenAI lib to do the requests?

```python
response = openai.Image.create(
    prompt='USER_PROMPT_GOES_HERE',
    size='1024x1024',
    n=1
)

image_url = response["data"][0]["url"]
```

This is an example for Azure itself.

I suppose `openai.Image.create` throws `OpenAIException` which you could handle via `handle_exceptions` method (the same way as it done for ChatCompletion calls).

Reply:
> aidial_adapter_openai.utils.exceptions.HTTPException 

I don't think this is a good idea due to the fact that we don't really know what the json model will be and how the Azure will decide to change it

Reply:
Yes, I tried to use the library from the very beginning, but the problem is that we did not have time to migrate to the 1.X openai python client. In our current version, dall-e 2 api is only supported